### ISSUE TITLE
XMLHttpRequest is not defined

### ISSUE THREAD

**Issue by kevinyao93** [hammerlab/pileup.js#499]
I'm completely lost on this subject - my code was working fine earlier, but after I rebuilt it with version 0.6.10 of pileup, I've been getting this message, I've tried rolling it back to 0.6.9, but it didn't seem to fix the issue. I've also tried manually installing xmlhttprequest into my environment but this issue still seems to come up. 
The code I'm trying to use to grab the data track is - 
```
    data: pileup.formats.twoBit({
      url: 'http://www.biodalliance.org/datasets/hg19.2bit',
    }),
```
I've been getting this error message - 
```
ReferenceError: XMLHttpRequest is not defined
     at RemoteFile.getFromNetwork (/usr/src/app/node_modules/pileup/dist/main/RemoteFile.js:122:21)
     at RemoteFile.getBytes (/usr/src/app/node_modules/pileup/dist/main/RemoteFile.js:53:19)
     at retryRemoteGet (/usr/src/app/node_modules/pileup/dist/main/data/TwoBit.js:191:21)
     at new TwoBit (/usr/src/app/node_modules/pileup/dist/main/data/TwoBit.js:214:5)
     at Object.create (/usr/src/app/node_modules/pileup/dist/main/sources/TwoBitDataSource.js:170:31)
     at Object../src/components/genome/GenomeLandingPage/GenomeLandingPage.js (/usr/src/app/src/components/genome/GenomeLandingPage/GenomeLandingPage.js:21:1)
     at __webpack_require__ (/usr/src/app/webpack/bootstrap e85fcf1b6189466d8b5b:637:1)
     at fn (/usr/src/app/webpack/bootstrap e85fcf1b6189466d8b5b:47:1)
     at Object../src/routes/genomebrowser/index.js (/usr/src/app/build/server.js:42281:95)
     at __webpack_require__ (/usr/src/app/webpack/bootstrap e85fcf1b6189466d8b5b:637:1)
```

Any help would be much appreciated

**Comment by akmorrow13** [comment]
That is odd, I am not sure why XMLHttpRequest would be missing. You can try installing it with node: https://stackoverflow.com/questions/32604460/xmlhttprequest-module-not-defined-found

**Comment by kevinyao93** [comment]
I've tried installing both the xmlhttprequest and xhr2 from npm and none of them see to be helping. I'm running my code on a docker enviornment, it was working fine earlier with pileup, but for some reason it doesn't seem to work at all anymore, here's my package.json if it helps
```

{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "engines": {
    "node": ">=6.5",
    "npm": ">=3.10"
  },
  "browserslist": [
    ">1%",
    "last 4 versions",
    "Firefox ESR",
    "not ie < 9"
  ],
  "dependencies": {
    "babel-polyfill": "^6.23.0",
    "bluebird": "^3.5.0",
    "body-parser": "^1.17.2",
    "classnames": "^2.2.5",
    "cookie-parser": "^1.4.3",
    "core-js": "^2.4.1",
    "d3": "^5.4.0",
    "distinct-colors": "^1.0.4",
    "express": "^4.15.3",
    "express-graphql": "^0.6.6",
    "express-jwt": "^5.3.0",
    "fixed-data-table-2": "^0.8.0",
    "graphql": "^0.10.3",
    "history": "^4.6.3",
    "isomorphic-style-loader": "^4.0.0",
    "js-sha1": "^0.6.0",
    "jsonwebtoken": "^7.4.1",
    "node-fetch": "^1.7.1",
    "normalize.css": "^7.0.0",
    "npm": "^5.8.0",
    "passport": "^0.4.0",
    "passport-facebook": "^2.1.1",
    "pileup": "^0.6.10",
    "pretty-error": "^2.1.1",
    "prop-types": "^15.5.10",
    "query-string": "^5.0.0",
    "react": "^15.5.4",
    "react-addons-shallow-compare": "^15.0.2",
    "react-autocomplete": "1.7.1",
    "react-bootstrap": "^0.31.2",
    "react-bootstrap-multiselect": "2.0.0",
    "react-cookie": "^0.4.7",
    "react-dimensions": "1.3.0",
    "react-dom": "^15.5.4",
    "react-js-pagination": "^3.0.1",
    "react-rangeslider": "2.1.0",
    "react-redux": "^5.0.2",
    "react-virtualized": "^9.18.0",
    "recharts": "^1.0.0-alpha.4",
    "redux": "^3.6.0",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.1.0",
    "sequelize": "^4.2.1",
    "serialize-javascript": "^1.3.0",
    "source-map-support": "^0.4.15",
    "sqlite3": "3.1.8",
    "universal-router": "^3.2.0",
    "whatwg-fetch": "^2.0.3",
    "xmlhttprequest": "^1.8.0"
  },
  "devDependencies": {
    "assets-webpack-plugin": "3.5.1",
    "autoprefixer": "^7.1.1",
    "babel-cli": "^6.24.1",
    "babel-core": "^6.25.0",
    "babel-eslint": "^7.2.3",
    "babel-jest": "^20.0.3",
    "babel-loader": "^7.1.1",
    "babel-plugin-rewire": "^1.1.0",
    "babel-preset-env": "^1.5.2",
    "babel-preset-react": "^6.24.1",
    "babel-preset-react-optimize": "^1.0.1",
    "babel-preset-stage-2": "^6.24.1",
    "babel-register": "^6.24.1",
    "babel-template": "^6.25.0",
    "babel-types": "^6.25.0",
    "browser-sync": "^2.18.12",
    "chokidar": "^1.7.0",
    "cross-env": "^5.0.1",
    "css-loader": "^0.28.4",
    "enzyme": "^2.9.1",
    "eslint": "^4.4.1",
    "eslint-config-airbnb": "^15.0.1",
    "eslint-config-prettier": "^2.3.0",
    "eslint-import-resolver-node": "^0.3.1",
    "eslint-loader": "^1.8.0",
    "eslint-plugin-css-modules": "^2.7.1",
    "eslint-plugin-flowtype": "^2.34.1",
    "eslint-plugin-import": "^2.6.1",
    "eslint-plugin-jsx-a11y": "^6.0.2",
    "eslint-plugin-prettier": "^2.1.2",
    "eslint-plugin-react": "^7.5.1",
    "file-loader": "^0.11.2",
    "front-matter": "^2.1.2",
    "glob": "^7.1.2",
    "husky": "^0.14.1",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^20.0.4",
    "jest-cli": "^20.0.4",
    "jest-codemods": "^0.12.0",
    "jscodeshift": "^0.3.32",
    "lint-staged": "^7.1.3",
    "markdown-it": "^8.3.1",
    "mkdirp": "^0.5.1",
    "null-loader": "^0.1.1",
    "opn-cli": "^3.1.0",
    "pixrem": "^4.0.1",
    "pleeease-filters": "^4.0.0",
    "postcss": "^6.0.4",
    "postcss-calc": "^6.0.0",
    "postcss-color-function": "^4.0.0",
    "postcss-custom-media": "^6.0.0",
    "postcss-custom-properties": "^6.1.0",
    "postcss-custom-selectors": "^4.0.1",
    "postcss-flexbugs-fixes": "^3.0.0",
    "postcss-import": "^10.0.0",
    "postcss-loader": "^2.0.6",
    "postcss-media-minmax": "^3.0.0",
    "postcss-nested": "^2.0.2",
    "postcss-nesting": "^4.0.1",
    "postcss-pseudoelements": "^5.0.0",
    "postcss-selector-matches": "^3.0.1",
    "postcss-selector-not": "^3.0.1",
    "prettier": "^1.13.4",
    "raw-loader": "^0.5.1",
    "react-deep-force-update": "^2.0.1",
    "react-error-overlay": "^1.0.9",
    "react-hot-loader": "^3.0.0-beta.7",
    "react-test-renderer": "^15.6.1",
    "rimraf": "^2.6.1",
    "sinon": "^3.2.0",
    "stylelint": "^8.0.0",
    "stylelint-config-standard": "^17.0.0",
    "stylelint-order": "^0.6.0",
    "svg-url-loader": "^2.1.1",
    "url-loader": "^0.5.9",
    "webpack": "^3.0.0",
    "webpack-bundle-analyzer": "^2.8.2",
    "webpack-dev-middleware": "^1.11.0",
    "webpack-hot-middleware": "^2.18.0",
    "webpack-node-externals": "^1.6.0",
    "xhr2": "0.1.4"
  },
  "babel": {
    "presets": [
      [
        "env",
        {
          "targets": {
            "node": "current"
          }
        }
      ],
      "stage-2",
      "react"
    ],
    "env": {
      "test": {
        "plugins": [
          "rewire"
        ]
      }
    },
    "ignore": [
      "/node_modules/",
      "/build/chunks/",
      "/build/public/",
      "/build/updates/",
      "/build/server.js"
    ]
  },
  "lint-staged": {
    "*.{js,jsx}": [
      "eslint --fix",
      "git add"
    ],
    "*.{json,graphql}": [
      "prettier --write",
      "git add"
    ],
    "*.{css,less,scss,sss}": [
      "stylelint --fix",
      "git add"
    ]
  },
  "scripts": {
    "precommit": "lint-staged && jest",
    "lint-js": "eslint --ignore-path .gitignore --ignore-pattern \"!**/.*\" .",
    "lint-css": "stylelint \"src/**/*.{css,less,scss,sss}\"",
    "lint": "yarn run lint-js && yarn run lint-css",
    "fix-js": "yarn run lint-js -- --fix",
    "fix-css": "yarn run lint-css -- --fix",
    "fix": "yarn run fix-js && yarn run fix-css",
    "test": "jest",
    "test-watch": "yarn run test -- --watch --notify",
    "test-cover": "yarn run test -- --coverage",
    "coverage": "yarn run test-cover && opn coverage/lcov-report/index.html",
    "clean": "babel-node tools/run clean",
    "copy": "babel-node tools/run copy",
    "bundle": "babel-node tools/run bundle",
    "build": "babel-node tools/run build",
    "build-stats": "yarn run build -- --release --analyse",
    "deploy": "babel-node tools/run deploy",
    "render": "babel-node tools/run render",
    "serve": "babel-node tools/run runServer",
    "start": "babel-node tools/run start"
  }
}
```

**Comment by kevinyao93** [comment]
Still couldn't figure out what the issue was, rebuilt the entire container and removed the xmlhttprequest and xhr2 packages, and it ended up resolving itself. Thanks for the help!
